File: PB - More Exercises JavaScript/6.0 Drawing Figures with Loops - More Exercises/05. Square Frame.js
function squareFrame(n) {
    //top
    console.log(`+${' -'.repeat(n - 2)} +`)
    
    //middle
    for (let i = 0; i < n - 2; i++) {
        console.log(`|${' -'.repeat(n - 2)} |`)
    }

    //bottom
    console.log(`+${' -'.repeat(n - 2)} +`)
}

// squareFrame(5)

File: PB - More Exercises GO/6.0 Drawing Figures with Loops - More Exercises/05. Square Frame.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var n int
	fmt.Scan(&n)
	//top
	fmt.Print("+", strings.Repeat(" -", n-2), " +\n")
	//middle
	for i := 0; i < n-2; i++ {
		fmt.Println("|" + strings.Repeat(" -", n-2) + " |")
	}
	//bottom
	fmt.Print("+", strings.Repeat(" -", n-2), " +\n")
}

/*
name   :05. Square Frame
input  :3
output :
+ - +
| - |
+ - +
*/
